pages/guides/build-a-cross-chain-voting-dapp/build-a-cross-chain-voting-dapp-part-1.mdx

```diff
@@ -0,0 +1,326 @@
+
+The folder structure of the project is shown in the following section. The project has a `contracts` folder which inside has a `jvm` and a `solidity` folder which hosts the Java and Solidity contracts and at the root of the project we have the main file `index.js` which hosts the logic to interact with the contracts and the xCall interface.
+
+```bash
+tree -I 'node_modules|build|local|test|bin|gradle'
+.
+├── config.js
+├── contracts
+│   ├── jvm
+│   │   ├── build.gradle
+│   │   ├── gradlew
+│   │   ├── gradlew.bat
+│   │   ├── settings.gradle
+│   │   └── VotingDapp
+│   │       ├── build.gradle
+│   │       └── src
+│   │           └── main
+│   │               ├── java
+│   │               │   └── app
+│   │               │       ├── RollbackData.java
+│   │               │       ├── VotingDapp.java
+│   │               │       └── XCallProxy.java
+│   │               └── resources
+│   └── solidity
+│       ├── contracts
+│       │   ├── interfaces
+│       │   │   ├── ICallServiceReceiver.sol
+│       │   │   └── ICallService.sol
+│       │   └── VotingDapp.sol
+│       └── migrations
+│           └── 1_voting_dapp.js
+├── deployments.json
+├── index.js
+├── lib.js
+├── package.json
+├── package-lock.json
+├── README.md
+├── truffle-config.js
+└── xcallAbi.json
+```
+
+## Java smart contract
+
+The smart contract on the origin chain in our setup is written in Java and for this example it will be deployed on the Berlin Testnet of the ICON Network.
+
+For setting up the Java development environment you can follow the instructions in the following tutorial:
+* https://icon.community/tutorials/java-tutorial-part-1-setting-development-environment-and-writing-smart-contract/
+
+The implementation of the smart contract will have two public (payable) methods called `voteYes()` and `voteNo()` which will trigger the crosschain transaction.
+
+These methods will internally call the private method `_sendCallMessage(byte[] _data, @Optional byte[] _rollback)`.
+
+The `_sendCallMessage` method in our Java contract will invoke the `sendCallMessage` method of the xCall contract on the origin chain to initiate the crosschain transfer.
+
+An internal tally of the votes that has been casted will be saved in two variables (BigInteger) that are named `countOfYes` and `countOfNo` and are initiated with a value of zero at the moment of deployment.
+
+The contract will have a `readonly` method called `getVotes()` that will return the current state of the votes which should be the same both on the destination and origin chains.
+
+During the deployment of this contract we are going to provide the address of the xcall contract on the origin chain and the btp address of the solidity contract on the destination chain which should be deployed first.
+
+
+The following is the entire contract written in Java, the code can be found in the `./contracts/jvm/VotingDapp/src/main/java/app/VotingDapp.java` file.
+
+```java
```

Comment:
All public methods should be documented in accordance with https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation

Reply:
done